### ISSUE TITLE
Configurable gestures

### ISSUE THREAD

**Issue by burakcan** [aeharding/voyager#64]
Apollo had configurable gestures and  I think it's a pretty useful thing to be able to customize. Screenshot from Apollo for reference: 

![image](https://github.com/aeharding/wefwef/assets/3121257/ae1090e2-c8bb-4095-bf83-d946a0bbf9e1)

**Comment by aeharding** [comment]
Should have options to disable left, right, and completely as well.

**Comment by rsammelson** [comment]
I'm going to work on this using the settings db PR as a base.

**Comment by rsammelson** [comment]
Hopefully will have this done on Friday, but it might be later in the weekend, the number of PRs I have open is getting a little hard to manage. Should clear up once the settings database is merged.

**Comment by aeharding** [comment]
@rsammelson, yeah, we really need to sort through the backlog of storage PRs lol

**Comment by rsammelson** [comment]
@aeharding still a little large on the open PR count, but I am going to start this.

Do you actually need to be logged in to hide posts, or can I remove that line in the process of adding configuration?

**Comment by rsammelson** [comment]
Also, how do mentions work? I can't get one to show up.

**Comment by rsammelson** [comment]
This is mostly done now except for a little debugging, and dealing with no swipe actions on one side.

**Comment by rsammelson** [comment]
I think this is pretty much done, but I branched off of #185 because my last commit there makes adding settings selectors much cleaner.

**Comment by robertafreese** [comment]
Not sure if you need this but in Apollo, there is a section in the "Gestures" panel in Settings, under the one in the above top comment screen shot to manager "Other" (which is not a great label, I would use something more action-oriented like "Control") so here is a screen shot below.

When either of the 'Disable" swipes is turned to On, that section that manages that swipe type is hidden. The control labeled Disable Navigation Gestures is very helpful (in my use case at least) when it is in the off position and both the left and right swipes are disabled. This use case makes swiping purely navigational, which helps when mobility is an issue. It keeps me from accidentally upvoting and replying and provides a purely gesture based navigation to the app. I can hit the voting and reply buttons if I want to do that, and I cannot accidentally vote or start a reply when what I wantto do is navigate. This is a huge usability advantage over most apps, I was accidentally upvoting and replying all the time before I found this screen, and this control just makes life a lot better.


![ApolloSettingsGesturesOther_issue64_Small](https://github.com/aeharding/voyager/assets/58523218/871bae39-b45e-42a5-ba0f-c5e35eed6ab3)

**Comment by robertafreese** [comment]
The "other" panel may actually be something relevant to #101

**Comment by jasperswallen** [comment]
@rsammelson I put together a branch that also implements this (https://github.com/jasperswallen/wefwef/tree/feat/swipe-options) - let me know if you’d prefer to use your implementation or if I should file a PR.

**Comment by rsammelson** [comment]
Based on a cursory look, I like my implementation based on two reasons: it's shorter 500 lines added vs 800, and I prefer having just one location that chooses the swipe action.

**Comment by rsammelson** [comment]
And for general information, I'm planning to convert that draft PR into a regular one in the next day or so once I get a chance to double check a couple things.

**Comment by robertafreese** [comment]
Is this supposed to have the ability to be a continuous back navigation?

If you tap into a post, then go to a community with the link on the name of the community, then go to a different post in the community, then go to a user profile with the link from a comment on that second post, can we expect swiping 'back' to undo that path?

Because I am finding that in this case the swipe quits working as navigation, when you get 'back' to the profile page, no spot anywhere will swipe back to the comment they made. 

It seems to be that you can only go 'back' two levels and then you can no longer swipe back and have to use the Back button.

Is this by design, something that Ionic is doing, or would this be something to put in a new issue? Possibly with a video of the steps?

**Comment by rsammelson** [comment]
That is not related to this issue, as this only changes the swipe actions in posts and comments and so forth, not navigation gestures. I would definitely make a new issue for that though.

**Comment by aeharding** [comment]
@robertafreese I believe that's #33

**Comment by robertafreese** [comment]
@aeharding got it, thanks!
